test/fixtures.ts

```diff
@@ -28,6 +27,15 @@ export const defaultService: Service<TestSchema> = {
   'GET /user_and_obj': (req, reply) => {
     return reply.status(200).send([{ name: 'user1' }, { id: '1', type: 'TEST' }, { any: 'thing' }]);
   },
+  'POST /jsonify': (req, reply) => {
+    const { date, regexp } = req.body;
+
+    return reply.status(200).send({
+      date: new Date(date),
```

Comment:
not possible to send just `date` as is even though it will still be a string on receiving end

part of my original intention was to allow sending data types compatible with JSON representation of declared type

while it would allow for sending just any strings, it could still be caught by fastify serializer because the schema is based on `Date` and is converted to `{type:string, format:'date-time'}`, I'm not sure if fastify does that by default, but the serializer can be configured

Reply:
I assume this is done in https://github.com/fastify/fast-json-stringify-compiler that uses https://github.com/fastify/fast-json-stringify/blob/307232f13f0f71cfbc4548082c587de940d80a48/index.js#L790-L797

If you require a particular format, why not enforce it using ﻿@format annotations on schema type also using string type instead can ensure that you do correct transform before `fastify-json-stringify` in case of complex types?

Reply:
In [b8716f6](https://github.com/Coobaha/typed-fastify/pull/92/commits/b8716f670ceb66b623b7d740697c3991493eef9c) I changed response body to be either `T` or `JsonValue`. So for date it is `Date | string`
<img width="634" alt="image" src="https://github.com/Coobaha/typed-fastify/assets/2446638/af08e9c0-3a88-4132-bbaf-d4b844dce176">

Reply:
> If you require a particular format, why not enforce it using ﻿@format annotations on schema type also using string type instead can ensure that you do correct transform before fastify-json-stringify in case of complex types?

Not 100% sure what you mean, but I'd assume the suggestion is to decalre `Schema` with `string` + `@format` annotation instead of relying on `Date` to get converted to corresponding JSONSchema. That's valid, but it would disable the option to provide `Date` in reply/request as a higher type, when declaring `Schema` with `Date` should allow sending either `Date` or `string`

____________________

Just replying for now, will have a look in the evening, thanks